src/wallet/use-cases/find-all/find-all.ts

```diff
@@ -0,0 +1,14 @@
+import { Injectable } from '@nestjs/common';
+
+import { IWalletRepository, IWalletsResponse } from '@/wallet/interfaces';
+
+@Injectable()
+export class FindAllWalletsUseCase {
```

Comment:
Qual a real necessidade de ter um useCase de findAll?

Digo isso, pois mesmo que tiver, deixo uma dúvida pra vc levar nas suas próximas implementações. As vezes, a gente acaba pensando em entregar mais coisa, mesmo que não seja usado, sendo que vc perde um pouco mais de tempo com coisas que realmente não vão fazer sentido pra aplicação como um todo.

Falo isso, pq no começo é normal a gente sair fazendo CRUD, achando que vai ser assim sempre. Lembre que vc precisa entregar o que o PO/Tech Lead escrevem no card, então é muito bom vc pensar além, mas pense se vale o esforço.


Se nesse caso aqui, realmente for necessário, só leva o comentário como aprendizado e reflexão haahha.

Reply:
A gente vai usar sem Levi! Mas realmente o comentário foi muito enriquecedor, irei começar a pensar dessa forma, obrigado!